src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -483,6 +483,12 @@ async def build_pex_component(
     constraint_file_digest = EMPTY_DIGEST
     requirements_file_digest = EMPTY_DIGEST
 
+    lockfile_scope_name: str | None
+    if isinstance(request.requirements, (ToolDefaultLockfile, ToolCustomLockfile)):
+        lockfile_scope_name = request.requirements.options_scope_name
+    else:
+        lockfile_scope_name = None
```

Comment:
As explained above, please call it `resolve_name`. Also generally prefer ternary expressions: https://www.pantsbuild.org/docs/style-guide#prefer-conditional-expressions-ternary-expressions

```python
resolve_name = (
     request.requirements.options_scope_name
     if isinstance(request.requirements, (ToolDefaultLockfile, ToolCustomLockfile))
     else None
)
```

In the future, this will need to capture the resolve name for multiple user lockfiles. It's fine to punt on that, but can you please add something like:

```
# TODO(#12314): Capture the resolve name for multiple user lockfiles.
```